File: src/android/ansight-core/src/main/kotlin/ai/ansight/runtime/purchases/PurchaseProduct.kt
package ai.ansight.runtime.purchases

import org.json.JSONObject

data class PurchaseProduct @JvmOverloads constructor(val productId: String, val available: Boolean,
    val observedAt: Long = System.currentTimeMillis(), val displayPrice: String? = null,
    val price: String? = null, val currencyCode: String? = null, val productType: String = "unknown") {
    fun toJson(): JSONObject = JSONObject().put("productId", productId).put("available", available)
        .put("observedAt", observedAt).put("displayPrice", displayPrice ?: JSONObject.NULL)
        .put("price", price ?: JSONObject.NULL).put("currencyCode", currencyCode ?: JSONObject.NULL).put("productType", productType)
    companion object {
        fun fromJson(value: JSONObject) = PurchaseProduct(value.getString("productId"), value.getBoolean("available"),
            if (value.has("observedAt")) value.getLong("observedAt") else System.currentTimeMillis(),
            if (value.isNull("displayPrice")) null else value.getString("displayPrice"),
            if (value.isNull("price")) null else value.getString("price"),
            if (value.isNull("currencyCode")) null else value.getString("currencyCode"), value.optString("productType", "unknown"))
    }
}

File: src/ios/Sources/AnsightCore/Purchases/PurchaseProduct.swift
import Foundation

public struct PurchaseProduct: Sendable, Codable, Equatable {
    public let productId: String
    public let available: Bool
    public let displayPrice: String?
    public let price: String?
    public let currencyCode: String?
    public let productType: String
    public let observedAt: Int64
    public init(productId: String, available: Bool, displayPrice: String? = nil, price: String? = nil, currencyCode: String? = nil, productType: String = "unknown", observedAt: Int64 = PurchaseDiagnostics.now()) {
        self.productId = productId; self.available = available; self.displayPrice = displayPrice
        self.price = price; self.currencyCode = currencyCode; self.productType = productType; self.observedAt = observedAt
    }
}
